datasets/ohsumed/ohsumed.py

```diff
@@ -0,0 +1,167 @@
+                "source": datasets.Value("string"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        # TODO: This method is tasked with downloading/extracting the data and defining the splits depending on the configuration
+        # If several configurations are possible (listed in BUILDER_CONFIGS), the configuration selected by the user is in self.config.name
+
+        # dl_manager is a datasets.download.DownloadManager that can be used to download and extract URLs
+        # It can accept any type or nested list/dict and will give back the same structure with the url replaced with path to local files.
+        # By default the archives will be extracted and a path to a cached folder where they are extracted is returned instead of the archive
+        my_urls = _URLs[self.config.name]
+        data_dir = dl_manager.download_and_extract(my_urls)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "train.jsonl"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={"filepath": os.path.join(data_dir, "test.jsonl"), "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: This method will receive as arguments the `gen_kwargs` defined in the previous `_split_generators` method.
+        # It is in charge of opening the given file and yielding (key, example) tuples from the dataset
+        # The key is not important, it's more here for legacy reason (legacy from tfds)
+        print(filepath)
+        with open(filepath, encoding="utf-8") as f:
+            abstracts = json.load(f)
+            for id_, data in enumerate(abstracts):
```

Comment:
Data seems to be in `jsonl` format, I would recommend using the following syntax to avoid load all the file in memory at once and take advantage of the generator pattern:
```suggestion
        with open(filepath, encoding="utf-8") as f:
            for id_, line in enumerate(abstracts):
                data = json.loads(line)
```

Reply:
this is more of a `json-list` rather than a` json-lines` file! 
To give you some background, the original dataset is in some weird format. I had to donwload the dataset and then processed it to a `jsonl` format.  You can check the pre-processing code in my repo here - https://github.com/skyprince999/ohsumed-preprocessed

Also refer to my discussion with @yjernite  on the slack channel 
https://huggingface.slack.com/archives/C01FSCW6JLU/p1607713806319200

The original format is something similar to this- 
"""
Row 1  Col 1
Row 1
Col 2
Row 1
Col 3
.
.
Row 2 Col 1
Row 2
Col 2 
Row 2
Col 3

"""

Reply:
Wow, thanks for taking the trouble to preprocess the data !
Would you mind refactoring to proper json-line files though ? I think it would be worth the hassle

Reply:
I loved the part that this is a 20 year old dataset (from pre 2000's era ...that's like before Y2k ! ) 

Done, dusted with the refactoring!

Reply:
currently the processed dataset is uploaded on my private gcp bucket. 
Is it possible to upload it to HF datahub ?

Reply:
Hi @skyprince999 !

Thank you so much for processing all the data - the format is indeed kinda weird.
Do you think it could be possible to include the processing directly inside the dataset script ?
I checked your notebook and this is typically the kind of processing we use in dataset scripts when formats are conventional.
This way the data can directly be downloaded from the trec website.

What do you think ?

Reply:
I believe this will be inside the `generate_examples` function...? do you have a sample which I can use as a reference?

Reply:
reviewing what will be required:
- integrate the data transformation logic 
- recreate the `dataset_info.json `file
- update the correct path to the extracted files
- update the dummy files similar to the extracted files

Reply:
> I believe this will be inside the generate_examples function...? do you have a sample which I can use as a reference?

Yes exactly. You can iterate over the lines and yield one example once its parsing is completed.

For example that's what the [conll2003](https://github.com/huggingface/datasets/blob/master/datasets/conll2003/conll2003.py) script does but for NER

Reply:
updated the new commit with the above steps. waiting for the tests to complete. 
 
It's quite late over here. so need to shut down for the night! :) 
will check in the morning.